File: haskell/pangram/src/Pangram.hs
module Pangram (isPangram) where

import Data.Char (toLower)

validCharacters :: String
validCharacters = "abcdefghijklmnopqrstuvwxyz"

isPangram :: String -> Bool
isPangram text = all (`elem` validCharacters) convertedText && allUsed convertedText validCharacters
  where
    convertedText = convert text

convert :: String -> String
convert text = filter (`elem` validCharacters) $ map toLower text

allUsed :: String -> String -> Bool
allUsed _ [] = True
allUsed [] _ = False
allUsed (x : xs) validText = allUsed xs (filter (/= x) validText)

File: cpp/pangram/pangram.cpp
#include "pangram.h"
#include <array>

namespace pangram
{

    bool is_pangram(const std::string &value)
    {
        const size_t size = 'z' - 'a' + 1;
        std::array<int, size> arr{};

        for (auto i : value)
        {
            int ch = tolower(i);

            if (ch >= 'a' && ch <= 'z')
            {
                arr[ch - 'a']++;
            }
        }

        for (auto i : arr)
        {
            if (i <= 0)
            {
                return false;
            }
        }

        return true;
    }

} // namespace pangram
